### ISSUE TITLE
Density test is failing with a panic on Jenkins

### ISSUE THREAD

**Issue by roberthbailey** [GoogleCloudPlatform/kubernetes#8695]
This is happening for both `Kubernetes e2e suite.Density [Performance suite] should allow starting 3 pods per node` and `Kubernetes e2e suite.Density [Performance suite] should allow starting 30 pods per node`. The output is Jenkins is

```
Stacktrace

/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:65
Test Panicked
```

@quinton-hoole @cjcullen

**Comment by bprashanth** [comment]
is it https://github.com/GoogleCloudPlatform/kubernetes/pull/7573#issuecomment-104754744 ?

**Comment by roberthbailey** [comment]
I think so. The offending code we found looked like:

```
resp, err := http.Get(c.Get().AbsPath(fmt.Sprintf("debug/pprof/%s", key)).URL().String() + "?debug=2")
body, err := ioutil.ReadAll(resp.Body)
```

So if the Get call returned an error, it was ignored and reading a nil response body would cause a panic.

**Comment by roberthbailey** [comment]
Fixed by #8691 (a revert).

@rrati FYI.
